internal/config/global.go

```
@@ -0,0 +1,182 @@
+		return err
+	}
+	b = append(b, '\n')
+
+	// Written to a sibling and renamed over the target. os.WriteFile truncates
+	// first, so an interrupted write would leave the config empty and every
+	// later run would fail to parse it until someone repaired the file by hand.
+	tmp, err := os.CreateTemp(filepath.Dir(g.path), ".config-*.json")
+	if err != nil {
+		return err
+	}
+	defer os.Remove(tmp.Name())
```

golangci-lint (2.12.2)

[error] Error return value of `os.Remove` is not checked

(errcheck)